server/src/main/kotlin/suwayomi/tachidesk/manga/impl/download/fileProvider/IFileDownloader.kt

```diff
@@ -0,0 +1,29 @@
+package suwayomi.tachidesk.manga.impl.download.fileProvider
+
+@FunctionalInterface
+interface IFileDownload {
+    suspend fun executeDownload(vararg args: Any): Boolean
+}
+
+@FunctionalInterface
+interface IFileDownload1<A> : IFileDownload {
```

Comment:
make up better names instead of numbering

Reply:
tbh, no clue how to name this, I added "Args" as an suffix
I'm open for better names